app/src/main/java/com/dkanada/gramophone/service/playback/LocalPlayer.java

```diff
@@ -37,6 +43,35 @@
 
     private PlaybackCallbacks callbacks;
 
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+    private interface MediaItemsCallbacks {
+        void onMediaItemsCreated(List<MediaItem> mediaItems, int position, int progress, boolean resetCurrentSong);
+    }
```

Comment:
I'm not a fan of the callback but I haven't looked at other solutions yet for an asynchronous operation on the executor service so I have no suggestions.

Reply:
I moved the callback logic back into the runnable for now. It makes more sense rn and I'll tackle the threading thing in a different pr later.